programs/server/Server.cpp

```diff
@@ -1486,16 +1486,21 @@ try
 
     /// Load global settings from default_profile and system_profile.
     global_context->setDefaultProfiles(config());
-    const Settings & settings = global_context->getSettingsRef();
 
     /// Initialize background executors after we load default_profile config.
     /// This is needed to load proper values of background_pool_size etc.
     global_context->initializeBackgroundExecutorsIfNeeded();
 
-    if (settings.async_insert_threads)
+    size_t async_insert_threads = config().getUInt("async_insert_threads", 16);
```

Comment:
I understand that it is read only here, so yeah, it could be just extracted from config inplace.
However, as for me, struct settings helps with annotating and describing possible settings.
Is it worth to be described in SERVER_SETTINGS?

Reply:
Thanks, I didn't know about `SERVER_SETTINGS` list.